File: src/net/liplum/registries/CioCLs.java
package net.liplum.registries;

import static  mindustry.graphics.CacheLayer.*;

import mindustry.graphics.CacheLayer;
import  net.liplum.ClientOnly;
public class CioCLs {
    @ClientOnly
    public static CacheLayer dynamicColor;

    @ClientOnly
    public static CacheLayer hologram;

    public static void load() {
        dynamicColor = new ShaderLayer(CioShaders.dynamicColor);
        hologram = new ShaderLayer(CioShaders.hologram);
    }
}

File: src/net/liplum/registries/CioCLs.kt
package net.liplum.registries

import mindustry.graphics.CacheLayer
import net.liplum.lib.shaders.SD

object CioCLs {
    private var registered = false
    private var allCacheLayers = ArrayList<CacheLayer>()
    var cyberion = CacheLayer.ShaderLayer(SD.Cyberion).register()
    @JvmStatic
    fun load() {
        allCacheLayers = ArrayList()
        registerAll()
    }
    @JvmStatic
    fun registerAll() {
        if (!registered) {
            CacheLayer.add(*allCacheLayers.toTypedArray())
            registered = true
        }
    }

    fun <T> T.register(): T where T : CacheLayer {
        allCacheLayers.add(this)
        return this
    }
}
